openedx/stanford/djangoapps/auth_lagunita/models.py

```diff
@@ -66,4 +73,40 @@ def need_subscribed(cls):
             Q(user__email__endswith='@example.com') |
             Q(user__email__endswith='.example.com')
         )
-        return infos
+        for info in infos:
+            if info.is_enrolled_in_any(blacklist):
+                log.info("Subscription blacklisted for user=%s", info.user.email)
+            else:
+                yield info
+
+    def is_enrolled_in_any(self, courses):
+        """
+        Check if user is enrolled in any of the specified courses
+        """
+        is_enrolled = CourseEnrollment.objects.filter(
+            user_id=self.user_id,
+            course_id__in=courses,
```

Comment:
Shouldn't we also filter on `is_active=True`, or are they blacklisted regardless?

Reply:
Hm, well in the first case where they regret or accidentally enrolled and then unenrolled on the first day, they still would have consciously checked the opt-in, so they would still want to get marketing emails, no?